src/main/java/com/spotify/github/v3/checks/Annotation.java

```diff
@@ -125,13 +125,24 @@ public interface Annotation {
    */
   @Value.Check
   @SuppressWarnings("checkstyle:magicnumber")
-  default void check() {
+  default Annotation check() {
```

Comment:
It seems confusing to me that a `check` method does both 1) throwing exceptions when data is invalid and 2) sanitizes data under some circumstances (by returning a new object).
This now requires the result of this method to be used at all times. 
I suggest we either a) also throw if if the case in L139 is true, or b) move this sanitization logic into a different method somehow.

Reply:
Oh, is using the return value supported natively by `immutables`? Then I guess this is fine after all. Sorry for the noise :)

Reply:
This is something immutables also recommends. https://immutables.github.io/immutable.html#normalization

The method name might be confusing. We can rename it to `validateAndNormalise`.
I also do not like `And` in a method name. But there is no other way to achieve this with immutables.